### PR TITLE
Updates for Nautobot 2.0

### LINKED ISSUE #88: Update Device Onboarding for Nautobot 2.0
###  

Update code to work with Nautobot 2.0

### PR DESCRIPTION
**Pull request by scetron**
### Nautobot 2.0 Related Changes:
- Minimum version updated to 2.0rc2.
- Moved Site to Location field - Changed model field name
- Moved DeviceRole to Role field - Changed Foreign Key reference
- Moved Slug references to Name fields
- Moved to Nautobot UI ViewSets
- Updated API Viewset and methods
- Updated Tests

Closes #88

### CODE REVIEW

File: nautobot_device_onboarding/api/serializers.py [hunk 1]
```diff
@@ -45,50 +31,18 @@ class OnboardingTaskSerializer(serializers.ModelSerializer):
...
+        slug_field="name",
     )
 
-    status = serializers.CharField(required=False, read_only=True, help_text="Onboarding Status")
-
-    failed_reason = serializers.CharField(required=False, read_only=True, help_text="Failure reason")
-
-    message = serializers.CharField(required=False, read_only=True, help_text="Status message")
-
     class Meta:  # noqa: D106 "Missing docstring in public nested class"
```
[the remark this answers is not in the document]
Comment (scetron) [reply nautobot_device_onboarding/api/serializers.py:40]: These are not actually removed, just not explicitly defined. They should be taken care of by the ValidatedModelSerializer.

E.G:
```
	"id": "9ce1ba31-ae46-4b64-b128-6bab0abc7e2c",
			"object_type": "nautobot_device_onboarding.onboardingtask",
			"display": "site | 172.20.20.8",
			"location": "site",
			"ip_address": "172.20.20.8",
			"port": 22,
			"timeout": 30,
			"role": null,
			"device_type": "",
			"platform": null,
			"created_device": null,
			"status": "failed",
			"failed_reason": "fail-connect",
			"message": "ERROR device unreachable: 172.20.20.8:22",
			"created": "2023-09-19T18:21:27.421173Z",
			"last_updated": "2023-09-19T18:21:27.461660Z"
```

File: development/docker-compose.dev.yml [hunk 2]
```diff
@@ -12,17 +12,17 @@ services:
...
-    ports:
-      - "8001:8080"
-    volumes:
-      - "../docs:/source/docs:ro"
-      - "../mkdocs.yml:/source/mkdocs.yml:ro"
-    image: "nautobot-device-onboarding/nautobot:${NAUTOBOT_VER}-py${PYTHON_VER}"
-    healthcheck:
-      disable: true
-    tty: true
+  # docs:
```
Comment (chadell) [inline development/docker-compose.dev.yml:15]: ??

**Review by chadell — COMMENTED** [review]
I guess this should NOT be merged into develop yet. I would establish a new branch like next-2.0 (you are actually using it), so I understand that the goal is not to merge but to review it and leave the branch as it is

File: nautobot_device_onboarding/api/serializers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (chadell) [reply nautobot_device_onboarding/api/serializers.py:40]: if they are already taken from there, why not the `location`?

File: nautobot_device_onboarding/nautobot_keeper.py [hunk 3]
```diff
@@ -290,19 +285,21 @@ def ensure_device_role(
...
-        except DeviceRole.DoesNotExist as err:
+            self.nb_device_role = Role.objects.get(name=self.netdev_nb_role_name)
+        except Role.DoesNotExist as err:
             if create_device_role:
-                self.nb_device_role = DeviceRole.objects.create(
-                    name=self.netdev_nb_role_slug,
-                    slug=self.netdev_nb_role_slug,
+                self.nb_device_role = Role.objects.create(
+                    name=self.netdev_nb_role_name,
+                    # slug=self.netdev_nb_role_name,
```
Comment (chadell) [inline nautobot_device_onboarding/nautobot_keeper.py:293]: some leftovers to clean up

File: nautobot_device_onboarding/navigation.py [hunk 4]
```diff
@@ -1,28 +1,39 @@
...
+                weight=1000,
+                items=(
+                    NavMenuItem(
+                        link="plugins:nautobot_device_onboarding:onboardingtask_list",
+                        name="Onboarding Tasks",
+                        permissions=["nautobot_device_onboarding.view_onboardingtask"],
+                        buttons=(
+                            NavMenuAddButton(
+                                link="plugins:nautobot_device_onboarding:onboardingtask_add",
+                                # name="Onboard",
```
Comment (chadell) [inline nautobot_device_onboarding/navigation.py:22]: why commented out?

File: nautobot_device_onboarding/templates/nautobot_device_onboarding/onboardingtask_retrieve.html [hunk 5]
```diff
@@ -0,0 +1,83 @@
+{% extends 'generic/object_detail.html' %}
```
Comment (chadell) [inline nautobot_device_onboarding/templates/nautobot_device_onboarding/onboardingtask_retrieve.html:1]: I think there is something wrong, as there is a copy from the old, and this that I guess it taking over?

File: nautobot_device_onboarding/tests/test_models.py [hunk 6]
```diff
@@ -14,50 +17,60 @@ class OnboardingDeviceModelTestCase(TestCase):
...
-        device_role = DeviceRole.objects.create(name="Firewall", slug="firewall")
-        device_type = DeviceType.objects.create(slug="srx3600", model="SRX3600", manufacturer=manufacturer)
+        status = Status.objects.get(name="Active")
+        location_type = LocationType.objects.create(name="site")
+
+        self.site = Location.objects.create(name="USWEST", location_type=location_type, status=status)
+        manufacturer = Manufacturer.objects.create(name="Juniper")
+        device_content_type = ContentType.objects.get_for_model(model=Device)
+        device_role = Role.objects.create(name="Firewall")
+        device_role.content_types.set([device_content_type])
```
Comment (chadell) [inline nautobot_device_onboarding/tests/test_models.py:27]: I always forget if the `set` needs a `save` or it's implicit (as it happens with `add`). just saying...

File: nautobot_device_onboarding/nautobot_keeper.py (hunk 3 above)
Comment (scetron) [reply nautobot_device_onboarding/nautobot_keeper.py:293]: Removed these and some others. Will check through all comments as well.

File: nautobot_device_onboarding/tests/test_models.py (hunk 6 above)
Comment (scetron) [reply nautobot_device_onboarding/tests/test_models.py:27]: I will double check this. Tests are passing for what it's worth :)

File: nautobot_device_onboarding/api/serializers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (scetron) [reply nautobot_device_onboarding/api/serializers.py:40]: location needs to have the name, without this it returns the pk.

File: nautobot_device_onboarding/templates/nautobot_device_onboarding/onboardingtask_retrieve.html (hunk 5 above)
Comment (scetron) [reply nautobot_device_onboarding/templates/nautobot_device_onboarding/onboardingtask_retrieve.html:1]: removed the copy. it was old and was used as reference.

**Review by chadell — APPROVED** [review]
LGTM (but do not merge to develop!)

**Pull request merged**